gen-snapshot-tests.js

```diff
@@ -0,0 +1,103 @@
+/**
+ * This script generates snapshot.test.js files from examples in docs.md files
+ * in every packages/wonder-blocks-* directory.
+ */
+const marked = require("marked");
+const fs = require("fs");
+const path = require("path");
+const unescape = require("unescape");
+const glob = require("glob");
+const babel = require("babel-core");
+const babylon = require("babylon");
+
+const styleguideConfig = require("./styleguide.config.js");
+
+function generateTestFile(root, examples, componentFileMap) {
+    const lines = [
+        "// This file is auto-generated by gen-tests.js",
+        `import React from "react";`,
+        `import renderer from "react-test-renderer"`,
+    ];
+
+    const modName = root.split("/")[1];
+
+    if (componentFileMap) {
+        for (const [componentName, filename] of Object.entries(componentFileMap)) {
+            const relFilename = path.relative(root, filename);
+            lines.push(`import ${componentName} from "./${relFilename}";`);
+        }
+    }
+
+    lines.push("");
+    lines.push(`describe("${modName}", () => {`);
+
+    examples.forEach((example, exampleIndex) => {
+        lines.push(`    it("example ${exampleIndex + 1}", () => {`);
+
+        const ast = babylon.parse(example, {
+            plugins: ["jsx", "flow"],
+        });   
+
+        const lastStatement = ast.program.body[ast.program.body.length - 1];
+
+        if (lastStatement.type !== "ExpressionStatement") {
+            throw new Error("last line should be an expression");
+        }
+
+        if (lastStatement.expression.type !== "JSXElement") {
+            throw new Error("last line should be a JSX Element");
+        }
+
+        // TODO(kevinb): add a `;` at the end of the last statement
```

Comment:
I wonder if we should generate the file and then just run it through prettier and let prettier take care of the formatting issues like this. You could also remove the extra leading spaces and let prettier format it correctly.

Reply:
I'll leave the prettier thing as a TODO since we don't have prettier set up for any of the files.